CMakeLists.txt

```diff
@@ -36,8 +36,21 @@ daq_add_unit_test(ProcessorInternalStateBufferManager_test LINK_LIBRARIES tpglib
 daq_add_unit_test(avx_processors_internal_state_collect_test LINK_LIBRARIES tpglibs)
 daq_add_unit_test(processor_state_registrate_collect_test LINK_LIBRARIES tpglibs)
 daq_add_unit_test(binary_signal_reader_test LINK_LIBRARIES tpglibs)
-
-# Add test application
+daq_add_unit_test(frame_reader_test LINK_LIBRARIES tpglibs)
+daq_add_unit_test(dummy_frame_adapter_test LINK_LIBRARIES tpglibs)
+daq_add_unit_test(tp_reader_test LINK_LIBRARIES tpglibs)
+daq_add_unit_test(tp_comparator_test LINK_LIBRARIES tpglibs)
+daq_add_unit_test(binary_file_validator_test LINK_LIBRARIES tpglibs)
+daq_add_unit_test(test_config_parser_test LINK_LIBRARIES tpglibs)
+
+# Add test applications
 daq_add_application(tpglibs_processor_test test_tpg_processor_app.cxx TEST LINK_LIBRARIES tpglibs)
+daq_add_application(tpglibs_generator_test test_tpg_generator_app.cxx TEST LINK_LIBRARIES tpglibs)
+
+# Add golden test data generator utility
+daq_add_application(generate_golden_test_data generate_golden_test_data.cxx TEST LINK_LIBRARIES tpglibs)
```

Comment:
"Golden test data" is a little, how to say, mysterious.
What is golden? What test?
I would rather talk about `reference` + `data` + `patterns`